tests/func/test_s3.py

```diff
@@ -45,6 +46,30 @@ def test_copy_singlepart_preserve_etag():
     RemoteS3._copy(s3, from_info, to_info, {})
 
 
+@mock_s3
+@pytest.mark.parametrize(
+    "base_info",
+    [RemoteS3.path_cls("s3://bucket/"), RemoteS3.path_cls("s3://bucket/ns/")],
+)
+def test_link_created_on_non_nested_path(base_info, tmp_dir, dvc, scm):
+    remote = RemoteS3(dvc, {"url": str(base_info.parent)})
+    remote.s3.create_bucket(Bucket=base_info.bucket)
+    remote.s3.put_object(
+        Bucket=base_info.bucket, Key=(base_info / "from").path, Body="data"
+    )
+    remote.link(base_info / "from", base_info / "to")
+
+    assert remote.exists(base_info / "from")
+    assert remote.exists(base_info / "to")
+
+
+@mock_s3
+def test_makedirs_doesnot_try_on_top_level_paths(tmp_dir, dvc, scm):
+    base_info = RemoteS3.path_cls("s3://bucket/")
+    remote = RemoteS3(dvc, {"url": str(base_info)})
+    remote.makedirs(base_info)
+
+
```

Comment:
Not testing the gs though :) And also maybe it is not worth the full blown func test (though don't want to waste it too). Maybe let's just add unit tests for RemoteS3/GS.makedirs methods that ensure that it doesn't try to create paths with empty `path`(pretty much same as `mkdir -p /` does nothing successfully)?

Reply:
Yeah, the test is not really related to the issue here.. But, as we are already here, I thought, we should at least cover our public apis (and, it's mocked s3 anyway).

Reply:
I'll add `s3` and `gs` related unittest for `makedirs`.